.github/workflows/release.yml

```
@@ -0,0 +1,225 @@
+        with:
+          fetch-depth: 0
+      - name: Create tag
+        run: |
+          VERSION="${{ needs.prepare.outputs.version }}"
+          BRANCH="${{ needs.prepare.outputs.branch }}"
+          if git ls-remote --tags origin "$VERSION" | grep -q "refs/tags/$VERSION"; then
+            echo "Tag $VERSION already exists. Skipping"; exit 0; fi
+          git fetch origin "$BRANCH":"$BRANCH"
+          git checkout "$BRANCH"
+          git tag "$VERSION"
+          git push origin "$VERSION"
+
+  publish-pypi:
+    name: Publish to PyPI
+    needs: [prepare, build, create-tag]
+    runs-on: ubuntu-latest
+    environment:
+      name: release
+      url: https://pypi.org/project/kubeflow/
+    steps:
+      - name: Download build artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: dist-${{ needs.prepare.outputs.version }}
+          path: dist/
+      - name: Publish to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          verbose: true
+
+  github-release:
+    name: Create GitHub Release
+    needs: [prepare, build, create-tag, publish-pypi]
+    runs-on: ubuntu-latest
+    environment:
+      name: release
+      url: https://github.com/kubeflow/sdk/releases
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Download build artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: dist-${{ needs.prepare.outputs.version }}
+          path: dist/
+      - name: Extract changelog
+        id: changelog
+        run: |
+          VERSION="${{ needs.prepare.outputs.version }}"
+          MAJOR_MINOR=$(echo "$VERSION" | cut -d. -f1,2)
+          CHANGELOG_FILE="CHANGELOG/CHANGELOG-${MAJOR_MINOR}.md"
+          if [[ -f "$CHANGELOG_FILE" ]]; then
+            set -euo pipefail
+            HEADER_REGEX="^# \\[${VERSION//./\\.}\\]"
+            SECTION=$(sed -n "/$HEADER_REGEX/,\$p" "$CHANGELOG_FILE" | tail -n +2)
+            NEXT_VERSION=$(echo "$SECTION" | grep -m1 "^# \\[[0-9]" || true)
+            if [[ -n "$NEXT_VERSION" ]]; then
+              CHANGELOG=$(echo "$SECTION" | sed -n "1,/^# \\[[0-9]/p" | sed '1d;$d')
+            else
+              CHANGELOG=$(echo "$SECTION" | sed '1d')
+            fi
+            if [[ -n "$CHANGELOG" ]]; then
+              {
+                echo "changelog<<EOF"
+                echo "$CHANGELOG"
+                echo "EOF"
+              } >> $GITHUB_OUTPUT
+            else
+              echo "changelog=**Note:** No changelog section found for $VERSION in $CHANGELOG_FILE" >> $GITHUB_OUTPUT
+            fi
+          else
+            echo "changelog=**Note:** $CHANGELOG_FILE not found" >> $GITHUB_OUTPUT
+          fi
+      - name: Create GitHub Release
+        uses: softprops/action-gh-release@v1
+        with:
+          tag_name: ${{ needs.prepare.outputs.version }}
+          name: Kubeflow SDK ${{ needs.prepare.outputs.version }}
```

Do we want to be consistent with naming of releases for [KF Trainer](https://github.com/kubeflow/trainer/releases/tag/v2.0.0) and [Llama Stack](https://github.com/llamastack/llama-stack/releases/tag/v0.2.21), and remove the project name from the release ?
```suggestion
          name: ${{ needs.prepare.outputs.version }}
```